app/forms/mappings/SelectLocalDateFormatter.scala

```diff
@@ -139,6 +141,20 @@ private[mappings] class SelectLocalDateFormatter(emptyMonthAndYearKey: String,
     }
   }
 
+  private def getDay(month: Int, year: Int) = {
+    val today = LocalDate.now
+    val todayMonth = today.getMonthValue
+    val todayYear = today.getYear
+    
+    val day: Int = (month, year) match {
+      case _ if useLastDayOfMonth => YearMonth.of(year, month).lengthOfMonth()
+      case (todayMonth, todayYear) =>
+        if((todayYear == year) && (todayMonth == month)) today.getDayOfMonth - 1 else 1
```

Comment:
is `if statement` necessary ? since its already handled in the `case statement`

Reply:
As it is written currently, `case _` is unreachable, so that if else is being used.

Reply:
Maybe just remove the `else 1` and put the `if` as a guard clause before `=>`

Reply:
I have corrected this